p2p/proto/consensus.proto

```diff
@@ -1,42 +1,121 @@
 syntax = "proto3";
 
-import "p2p/proto/common.proto";
-import "p2p/proto/transaction.proto";
-import "p2p/proto/state.proto";
+import "common.proto";
 import "google/protobuf/timestamp.proto";
 
+package consensus;
+
 
 // WIP - will change
 
-message Proposal {
-    uint64                    block_number      = 2;
-    uint32                    round             = 3;
-    uint32                    pol               = 4;  // proof of lock
-    Hash                      block_header_hash = 5;
-    google.protobuf.Timestamp timestamp         = 6;
-    ConsensusSignature        signature         = 7;
+message Transaction {
+    bytes elements = 1;
+}
+
+message TransactionBatch {
+    repeated Transaction transactions = 1;
+}
+
+message Proof {
+    bytes elements = 1;
+}
+
+message BlockNumber {
+    Hash   parent_hash = 1;
```

Comment:
I'll include the fork_id question here. BlockNumber is about identifying where in the blockchain a block lies.

I agree that `height` and `parent_hash` are basically equivalent. I just thought it was nice to have them both explicitly.

`fork_id` though is used in the case that there is a fork, when the L1 contract resets the validator set. If the L1 contract has finalized Starknet block `H` when it forks, it is possible that Starknet already decided within L2 block `H+1`. Starknet now needs to return to height `H+1` because of the L1 fork. We need a way to disambiguate the `H+1` from before the fork and the `H+1` from after the fork.